### DISCUSSION TITLE
consult-line can't find input word

### DISCUSSION THREAD

**Discussion by mmqqdd** [minad/consult#879] [Q&A] [answered]
"The consult-line function only matches strings that start with a specific sequence. For example, if there's a line starting with "hello" in the text, I can find it by inputting hello in consult-line. However, if hello appears in the middle of a line, it can't be found. Why is this?"

consult-line-multi   ans   consult-ripgrep  is  ok
<img width="1176" alt="image" src="https://github.com/minad/consult/assets/49719057/30d7af9b-19e2-4af0-91e0-6f3d0ee72b1b">

**Comment by oantolin** [comment, marked as answer]
Which lines match your input depends on your choice of `completion-styles`. The behavior you describe sounds like you have kept the default value of that variable. Try adding some style that doesn't only match at the beginning, such as `substring` or `flex` that come with Emacs, or `orderless` from the orderless package.

**Reply by mmqqdd** [reply]
Thank you very much for your help, I solved the problem by setting completion-styles
